### PR TITLE
Pull Request for Issue2162: PGM Components - Energy

### PR DESCRIPTION
**Pull request by dleacock**
Simple AMPVwStatusControl for energy, added editor to persistent view.

ToDo:

- [x] Find PV and add it.

### CODE REVIEW

**Comment by dleacock** [comment]
I'm getting an error while trying to read this PV in my control editor. I'm missing a "stopPV" which could be the reason why.  Waiting till I have access to PGM controls to resolve issue.

**Comment by dretrex** [comment]
That's because you didn't provide a stop PV in your constructor of the energy.

**Comment by dleacock** [comment]
Yes, thank you. I put that in my comment above. I couldn't find the pv on the pgm controls, nor the pv database but next monday I'll have another look.

**Comment by dleacock** [comment]
Undulator

ToDo:
- [ ] Widgets: Tracking button, Energy (actual value), Gap (mm) Set Point and actual value.  "At Gap" and "Moving" light.
- [ ] PV's

File: source/beamline/PGM/PGMBeamline.cpp [hunk 1]
```diff
@@ -67,6 +69,7 @@ void PGMBeamline::setupMono()
 
 void PGMBeamline::setupComponents()
 {
+    energy_ = new AMPVwStatusControl("Energy", "BL1611-ID-2:Energy:fbk", "BL1611-ID-2:Energy", "BL1611-ID-2:status", "BL1611-ID-2:emergStop", this, 0.001, 2.0, new CLSMAXvControlStatusChecker());
```
Comment (XuetaoLiu-CLS) [inline source/beamline/PGM/PGMBeamline.cpp]: @dleacock `BL1611-ID-2:emergStop` is not the stop PV for energy.

`BL1611-ID-2:emergStop` will try to stop a bunch of Beamline motors, including mono, PSL16114I2x01, SMTR16113I2010:emergStop, SMTR16113I2011:emergStop, SMTR1411-02:stop, FM16113I2001:emergStop, VAM16113I2001:emergStop PP, M16113I2001:emergStop PP, M16113I2002:emergStop, BL1611-ID-2:emergStop2.PROC.

I guess the  closest one is probably `PGM_mono:emergStop`, which will try to stop the mono motors: `SMTR16114I2001:emergStop`, `SMTR16114I2002:emergStop`, `SMTR16114I2003:emergStop` and `SMTR16114I2004:stop`

**Comment by dretrex** [comment]
Time Estimate: 1 minute

**Pull request merged**